### PR TITLE
Update weaver dependencies and repository links

### PR DESCRIPTION
**Pull request by sandeepnRES**
* chore: update all go module dependencies, add all go tests to workflow
* fix(weaver-go-sdk): use updated weaver protobuf for fabric and corda view
* chore: add script to generate checksums for go modules
* fix: update broken corda repository links
* ci: fix besu asset exchange workflow to use node 16

### CODE REVIEW

**Comment by VRamakrishna** [comment]
Looks like we are using Node 16 (as per the diff) and not Node 18 (as per the above comment). Right?

File: core/network/fabric-interop-cc/libs/testutils/mocks/chaincodestub.go [hunk 1]
```diff
@@ -4,9 +4,9 @@ package mocks
 import (
 	"sync"
 
-	"github.com/golang/protobuf/ptypes/timestamp"
 	"github.com/hyperledger/fabric-chaincode-go/shim"
 	"github.com/hyperledger/fabric-protos-go/peer"
+    timestamp "google.golang.org/protobuf/types/known/timestamppb"
```
Comment (VRamakrishna) [inline core/network/fabric-interop-cc/libs/testutils/mocks/chaincodestub.go:9]: Fix indent

File: go-mod-tidy.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ROOT_DIR=${1:-'..'}
+
+GOMODULE_PATHS=("core/network/fabric-interop-cc/libs/utils"
+"core/network/fabric-interop-cc/libs/assetexchange"
+"core/network/fabric-interop-cc/interfaces/asset-mgmt"
+"core/network/fabric-interop-cc/contracts/interop"
+"sdks/fabric/go-sdk"
+"samples/fabric/go-cli"
+"samples/fabric/simpleasset"
+"samples/fabric/satpsimpleasset"
```
Comment (VRamakrishna) [inline go-mod-tidy.sh:13]: Remove this. The `satp` code is not part of Weaver.

File: samples/fabric/simplestate/go.mod [hunk 3]
```diff
@@ -3,40 +3,37 @@ module github.com/hyperledger-labs/weaver-dlt-interoperability/samples/fabric/si
...
+	golang.org/x/net v0.23.0 // indirect
+	golang.org/x/sys v0.19.0 // indirect
+	golang.org/x/text v0.14.0 // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20240415180920-8c6c420018be // indirect
+	google.golang.org/grpc v1.63.2 // indirect
+	google.golang.org/protobuf v1.34.1 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
-
-replace github.com/hyperledger-labs/weaver-dlt-interoperability/core/network/fabric-interop-cc/libs/testutils => ../../../core/network/fabric-interop-cc/libs/testutils/
```
Comment (VRamakrishna) [inline samples/fabric/simplestate/go.mod:42]: Good catch!

**Review by VRamakrishna — CHANGES_REQUESTED** [review]
Just a couple of minor fixes required.

File: go-mod-tidy.sh (hunk 2 above)
Comment (sandeepnRES) [reply go-mod-tidy.sh]: Done.

File: core/network/fabric-interop-cc/libs/testutils/mocks/chaincodestub.go (hunk 1 above)
Comment (sandeepnRES) [reply core/network/fabric-interop-cc/libs/testutils/mocks/chaincodestub.go]: Done.

**Comment by sandeepnRES** [comment]
> Looks like we are using Node 16 (as per the diff) and not Node 18 (as per the above comment). Right?

Yes we are using Node 16, sorry that was old description, just now edited the PR desciption.

**Review by VRamakrishna — APPROVED** [review]

**Pull request merged**